Add instructor note about learning outcomes (#554)
carpentries/lesson-development-training: episodes/objectives.md

## Learning Objectives

::::::::::::::::::::::::::::::::::::::::::::::::: instructor

### Outcomes and Objectives

> In our experience, training participants sometimes have a strong preference for talking about _outcomes_ rather than _objectives_.

Reviewer: ```suggestion
In our experience, training participants sometimes have a strong preference for the phrase learning _outcomes_ or learning _objectives_.
```
